model/common/src/icon4py/model/common/grid/base.py

```diff
@@ -127,20 +140,20 @@ def has_skip_values(self):
         return self.config.limited_area or self.geometry_type == GeometryType.ICOSAHEDRON
 
     @functools.cached_property
-    def offset_providers(self):
-        offset_providers = {}
-        for key, value in self.offset_provider_mapping.items():
+    def connectivities(self) -> Dict[str, gtx.Connectivity]:
+        connectivity_map = {}
+        for key, value in self._connectivity_mapping.items():
             try:
                 method, *args = value
-                offset_providers[key] = method(*args) if args else method()
+                connectivity_map[key] = method(*args) if args else method()
             except MissingConnectivity:
                 warnings.warn(f"{key} connectivity is missing from grid.", stacklevel=2)
 
-        return offset_providers
+        return connectivity_map
 
     @utils.chainable
-    def with_connectivities(self, connectivity: Dict[gtx.Dimension, data_alloc.NDArray]):
-        self.connectivities.update({d: k.astype(gtx.int32) for d, k in connectivity.items()})
+    def with_neighbor_tables(self, connectivity: Dict[gtx.Dimension, data_alloc.NDArray]):
```

Comment:
Since this is updating the object in place, I would rename  it to:
```suggestion
    def set_neighbor_tables(self, connectivity: Dict[gtx.Dimension, data_alloc.NDArray]):
```
and the same applies to similar methods in this file.